retrieval/discovery/marathon/url.go

```diff
@@ -0,0 +1,14 @@
+package marathon
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+const appsURLFormat string = "http://%s/v2/apps/?embed=apps.tasks"
+
+// RandomAppsURL randomly selects a server from an array and creates an URL pointing to the app list
+func RandomAppsURL(servers []string) string {
```

Comment:
Take the new etcd client library as an example: http://godoc.org/github.com/coreos/etcd/client#Client

The client is able to sync the list of servers it can send its requests to. So the list you create your client with only has to be valid initially. Infrastructure changes, which is probably also true for Marathon, i.e. moving Marathon nodes would _ideally_ not break a running Prometheus server.

I'd just like to have a TODO as a reminder to check whether the Marathon API ever supports such behavior in the future (by simply giving us a list of currently reachable nodes).